src/components/PostsList.tsx

```diff
@@ -1,85 +1,95 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Post } from '../types/Post';
+import { PostContext } from './context/PostContext';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  posts: Post [],
+  selectedPost: Post | null,
+  setSelectedPost: React.Dispatch<React.SetStateAction<Post | null>>,
+  setIsCommentFormActive: React.Dispatch<React.SetStateAction<boolean>>,
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = React.memo(({
+  posts,
+  selectedPost,
+  setSelectedPost,
+  setIsCommentFormActive,
+}) => {
+  const [isFirstClick, setIsFirstClick] = useState(false);
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+  const {
+    setIsLoading,
+  } = React.useContext(PostContext);
```

Comment:
You are using the `setIsLoading` function from the PostContext, but you have not provided any context value to the PostsList component. Ensure that you wrap PostsList with a Provider for PostContext.